### ISSUE TITLE
Failed to update to latest patch version

### ISSUE THREAD

**Issue by ersoma** [shivammathur/setup-php#631]
**Describe the bug**
It looks like `shivammathur/setup-php@v2` is not respecting the patch version specified in the php-version variable nor the update parameter if it is set to true. Currently I couldn't found a way to enforce PHP version 8.1.9 on GitHub Actions. Also, some additional logs would be great to make it quickly noticeable that the version set up by the action is not the exact one specified by the parameter. (Also tried just setting the version to 8.1 with update: true, still 8.1.8 was selected)

**Version**
- [x] `v2`
- [ ] `v1`

**Runners**
- [x] GitHub Hosted
- [ ] Self Hosted

**Operating systems**
`runs-on: ubuntu-latest`

**PHP versions**
8.1.9

**To Reproduce**
```
      - name: Setup PHP 8.1.9 with composer
        uses: shivammathur/setup-php@v2
        with:
          php-version: 8.1.9
          tools: composer
        env:
          update: true
```

**Expected behavior**
```
==> Setup PHP
✓ PHP Found PHP 8.1.9

==> Setup Tools
✓ composer Added composer 2.3.10
```

**Screenshots/Logs**
```
==> Setup PHP
✓ PHP Found PHP 8.1.8

==> Setup Tools
✓ composer Added composer 2.3.10
```

**Additional context**

**Are you willing to submit a PR?**
No

**Comment by shivammathur** [comment]
The GitHub Actions runner images and `setup-php` both use builds from [`ppa:ondrej/php`](https://launchpad.net/~ondrej/+archive/ubuntu/php). There is no build for `8.1.9` yet on the ubuntu PPA.

https://github.com/oerdnj/deb.sury.org/issues/1821

**Comment by ersoma** [comment]
Thank you for the quick reply, issue is definitely not a bug than.
A few suggestions to consider:
- Maybe add it to the readme where the PHP version is coming from. I found the package now at the end but this was not obvious for me when I looked for the PHP source.
- Maybe add a comment to the GitHub Action's output in case the specified PHP version could not be exactly fulfilled, so it's more obvious the problem is not with the action itself or how it was used.

Thanks!
